src/editor/Editor.tsx

```diff
@@ -27,38 +28,62 @@ const specialCharacterKeys = Object.entries(specialCharacters).map(
     }
   })
 )
+interface Props {
+  readonly name: string
+  readonly value: string
+  readonly onChange: (value: string, event?: any) => void | undefined
+  readonly disabled: boolean
+  readonly error: object
+}
+
+class Editor extends Component<Props> {
```

Comment:
Since the introduction of [hooks ](https://reactjs.org/docs/hooks-intro.html) refs etc. can also be used in function components.